server/core/runtime/steps_runner.go

```diff
@@ -23,7 +23,7 @@ func NewStepsRunner(
 	versionStepRunner Runner,
 	runStepRunner CustomRunner,
 	envStepRunner EnvRunner,
-) *stepsRunner {
```

Comment:
nit: should we also add a comment on why we're skipping the lint check here and in general when we do?

Reply:
i only added it to simplify adding this action, otherwise I'd need to refactor a lot of things completely.  In this case, we shouldn't be returning an unexported struct from an exported function.  So going forward we will not be doing that.

In the future we should add a comment when we do it and add the specific reason, I just didn't do it here because there were too many places to do that.